src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports System.Globalization
+Imports System.Runtime.InteropServices
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Collections
+Imports Microsoft.CodeAnalysis.RuntimeMembers
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend Class TupleTypeSymbol
+        Inherits WrappedNamedTypeSymbol
+
+        Private _locations As ImmutableArray(Of Location)
+
+        Private _elementLocations As ImmutableArray(Of Location)
+
+        Private _elementNames As ImmutableArray(Of String)
+
+        Private _elementTypes As ImmutableArray(Of TypeSymbol)
+
+        Private _lazyMembers As ImmutableArray(Of Symbol)
+
+        Private _lazyFields As ImmutableArray(Of FieldSymbol)
+
+        Private _lazyUnderlyingDefinitionToMemberMap As SmallDictionary(Of Symbol, Symbol)
+
+        Friend Const RestPosition As Integer = 8
+
+        Friend Const TupleTypeName As String = "ValueTuple"
```

Comment:
will enter a tracking issue on this. Tuples may end up with some common code. Makes sense to factor it out.

---
In reply to: [71560501](https://github.com/dotnet/roslyn/pull/12596#discussion_r71560501) [](ancestors = 71560501)